File: src/main/java/pro/evanwright/saphira/exception/UncheckedSQLException.java
package pro.evanwright.saphira.exception;

/**
 * A wrapped, unchecked {@link java.sql.SQLException}
 */
public class UncheckedSQLException extends RuntimeException {
    public UncheckedSQLException(Throwable throwable) {
        super(throwable);
    }

    public UncheckedSQLException(String errorMessage, Throwable throwable) {
        super(errorMessage, throwable);
    }
}

File: src/main/java/pro/evanwright/saphira/exception/UncheckedSQLException.kt
package pro.evanwright.saphira.exception

/**
 * A wrapped, unchecked [java.sql.SQLException] in an attempt to make the code cleaner
 * and give more flexibility to the user of the API.
 */
class UncheckedSQLException : RuntimeException {
    constructor(throwable: Throwable?) : super(throwable) {}
    constructor(errorMessage: String?, throwable: Throwable?) : super(errorMessage, throwable) {}
}